StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1208SystemUsingDirectivesMustBePlacedBeforeOtherUsingDirectives.cs

```diff
@@ -50,93 +53,79 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            context.RegisterSyntaxNodeActionHonorExclusions(this.HandleUsingDirectiveSyntax, SyntaxKind.UsingDirective);
+            context.RegisterSyntaxNodeActionHonorExclusions(this.HandleCompilationUnit, SyntaxKind.CompilationUnit);
+            context.RegisterSyntaxNodeActionHonorExclusions(this.HandleNamespaceDeclaration, SyntaxKind.NamespaceDeclaration);
         }
 
-        private void HandleUsingDirectiveSyntax(SyntaxNodeAnalysisContext context)
+        private void HandleCompilationUnit(SyntaxNodeAnalysisContext context)
         {
-            UsingDirectiveSyntax syntax = context.Node as UsingDirectiveSyntax;
-            if (syntax.Alias != null)
-            {
-                return;
-            }
+            var compilationUnit = context.Node as CompilationUnitSyntax;
 
-            SemanticModel semanticModel = context.SemanticModel;
-            INamespaceSymbol namespaceSymbol;
-            string topLevelNamespace = GetTopLevelNamespace(semanticModel, syntax, out namespaceSymbol, context.CancellationToken);
-            if (!"System".Equals(topLevelNamespace, StringComparison.Ordinal))
-            {
-                return;
-            }
+            var usings = compilationUnit.Usings;
 
-            CompilationUnitSyntax compilationUnit = syntax.Parent as CompilationUnitSyntax;
-            SyntaxList<UsingDirectiveSyntax>? usingDirectives = compilationUnit?.Usings;
-            if (!usingDirectives.HasValue)
-            {
-                NamespaceDeclarationSyntax namespaceDeclaration = syntax.Parent as NamespaceDeclarationSyntax;
-                usingDirectives = namespaceDeclaration?.Usings;
-            }
+           ProcessUsingsAndReportDiagnostic(usings, context);
+        }
+
+        private void HandleNamespaceDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            var namespaceDeclaration = context.Node as NamespaceDeclarationSyntax;
+
+            var usings = namespaceDeclaration.Usings;
+
+            ProcessUsingsAndReportDiagnostic(usings, context);
+        }
 
-            if (!usingDirectives.HasValue)
+        private static void ProcessUsingsAndReportDiagnostic(SyntaxList<UsingDirectiveSyntax> usings, SyntaxNodeAnalysisContext context)
+        {
+            if (!usings.Any())
             {
                 return;
             }
 
-            foreach (var usingDirective in usingDirectives)
+            string systemUsingDirectivesShouldBeBeforeThisName = null;
+            for (var i = 1; i < usings.Count; i++)
             {
-                // we are only interested in nodes before the current node
-                if (usingDirective == syntax)
-                {
-                    break;
-                }
+                var @using = usings[i];
```

Comment:
You could call this variable usingDirective to get rid of the verbatim identifier.